### PR TITLE
Azeventhubs go amqp

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Testing the updated go-amqp public surface area.

### CODE REVIEW

File: sdk/messaging/azeventhubs/internal/links.go [hunk 1]
```diff
@@ -33,7 +32,7 @@ type LinkWithID[LinkT AMQPLink] struct {
 // (for unit testing only)
 type LinksForPartitionClient[LinkT AMQPLink] interface {
 	RecoverIfNeeded(ctx context.Context, partitionID string, lwid *LinkWithID[LinkT], err error) error
-	Retry(ctx context.Context, eventName log.Event, operation string, partitionID string, retryOptions exported.RetryOptions, fn func(ctx context.Context, lwid LinkWithID[LinkT]) error) error
+	Retry(ctx context.Context, eventName azlog.Event, operation string, partitionID string, retryOptions exported.RetryOptions, fn func(ctx context.Context, lwid LinkWithID[LinkT]) error) error
```
Comment (richardpark-msft) [inline sdk/messaging/azeventhubs/internal/links.go:35]: I don't mind it, but why did this need to change?

File: sdk/messaging/azeventhubs/internal/errors.go [hunk 2]
```diff
@@ -176,15 +176,16 @@ func GetRecoveryKind(err error) RecoveryKind {
 	}
 
 	// check the "special" AMQP errors that aren't condition-based.
-	if errors.Is(err, amqp.ErrLinkClosed) || IsQuickRecoveryError(err) {
+	if IsQuickRecoveryError(err) {
```
Comment (richardpark-msft) [inline sdk/messaging/azeventhubs/internal/errors.go:179]: It's nice that this simplified. One less error type to check for the same condition.

**Review by richardpark-msft — APPROVED** [review]
LGTM!

**Comment by richardpark-msft** [comment]
> 

Trust eroded. :) Did you want to merge before those changes or do you think this work is potentially invalid without it? I wasn't reviewing the go-amqp side.

**Review by richardpark-msft — CHANGES_REQUESTED** [review]
There's some stuff in here that's not quite ready to merge yet (as per our earlier discussion)

**Comment by jhendrixMSFT** [comment]
This PR is still valid as it updates to the `v0.18.0` public surface area changes.  I'd just need to roll back the changes post that.

File: sdk/messaging/azeventhubs/internal/links.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/messaging/azeventhubs/internal/links.go:35]: hmm yes it consolidated the duplicate imports

**Pull request closed without merging**